core/sys/pt.h

```diff
@@ -296,6 +296,34 @@ struct pt {
   } while(0)
 
 /**
+ * \brief Yield from the current protothread then execute a statement
+ *
+ * This function will yield the protothread, thereby allowing other
+ * processing to take place in the system.
+ *
+ * This macro is intended to be used when there is a possibility that the
+ * given statement will cause recursion into the protothread either directly
+ * or through an ISR. Without this construction, the protothread would not
+ * correctly continue from the yield point.
+ *
+ * Note that proto-threads are never truly reentrent.
+ *
+ * \param pt A pointer to the protothread control structure.
+ * \param stmnt code statement to execute before yielding
+ *
+ * \hideinitializer
+ */
+#define PT_YIELD_ON(pt,stmnt)                         \
+  do{                                                 \
+    PT_YIELD_FLAG = 0;                                \
+    LC_SET((pt)->lc);                                 \
+    if(PT_YIELD_FLAG == 0) {                          \
+      { stmnt ;}                                      \
+      return PT_YIELDED;                              \
+    }                                                 \
+  }while(0)
```

Comment:
> I still don't like the name. ```PT_YIELD_FOR()```, while not perfect, seems, to me, to be more appropriate.

```PT_YIELD_ON``` makes sense to me because the macro yields 'on' the given statement such that the proto-thread's execution 'pointer' is made to point just after the statement.